pkg/costmodel/costmodel.go

```diff
@@ -134,42 +134,118 @@ func (cd *CostData) GetController() (name string, kind string, hasController boo
+	if err != nil {
+		log.Warnf("ComputeCostData: continuing despite prometheus errors: %s", err)
+	}
+
+	defer measureTime(time.Now(), profileThreshold, "ComputeCostData: Processing Query Data")
 
-	podDeploymentsMapping, err := getPodDeployments(cm.Cache, podlist, clusterID)
+	nodes, err := cm.GetNodeCost()
 	if err != nil {
+		log.Warnf("GetNodeCost: no node cost model available: %s", err)
 		return nil, err
 	}
 
-	podServicesMapping, err := getPodServices(cm.Cache, podlist, clusterID)
+	// Unmounted PVs represent the PVs that are not mounted or tied to a volume on a container
+	unmountedPVs := make(map[string][]*PersistentVolumeClaimData)
+	pvClaimMapping, err := GetPVInfoLocal(cm.Cache, clusterID)
 	if err != nil {
-		return nil, err
+		log.Warnf("GetPVInfo: unable to get PV data: %s", err.Error())
+	}
+	if pvClaimMapping != nil {
+		err = cm.addPVData(pvClaimMapping)
+		if err != nil {
+			return nil, err
+		}
+		// copy claim mappings into zombies, then remove as they're discovered
+		for k, v := range pvClaimMapping {
+			unmountedPVs[k] = []*PersistentVolumeClaimData{v}
+		}
 	}
 
-	namespaceLabelsMapping, err := getNamespaceLabels(cm.Cache, clusterID)
+	networkUsageMap, err := GetNetworkUsageData(resNetZoneRequests, resNetRegionRequests, resNetInternetRequests, clusterID)
 	if err != nil {
-		return nil, err
+		log.Warnf("Unable to get Network Cost Data: %s", err.Error())
+		networkUsageMap = make(map[string]*NetworkUsageData)
 	}
 
-	namespaceAnnotationsMapping, err := getNamespaceAnnotations(cm.Cache, clusterID)
+	containerNameCost, missingNodes, missingContainers, err := getContainerCostData(podlist, resRAMUsage, resCPUUsage, pvClaimMapping, unmountedPVs, networkUsageMap, podDeploymentsMapping, podServicesMapping, namespaceLabelsMapping, namespaceAnnotationsMapping, nodes, cm.ClusterMap, cp, clusterID)
 	if err != nil {
 		return nil, err
 	}
 
+	// Use unmounted pvs to create a mapping of "Unmounted-<Namespace>" containers
+	// to pass along the cost data
+	unmounted := findUnmountedPVCostData(cm.ClusterMap, unmountedPVs, namespaceLabelsMapping, namespaceAnnotationsMapping)
+	for k, costs := range unmounted {
+		log.Debugf("Unmounted PVs in Namespace/ClusterID: %s/%s", costs.Namespace, costs.ClusterID)
+
+		containerNameCost[k] = costs
+	}
+
+	err = findDeletedNodeInfo(cm.DataSource, missingNodes, start, end)
+	if err != nil {
+		log.Errorf("Error fetching historical node data: %s", err.Error())
+	}
+
+	err = findDeletedPodInfo(cm.DataSource, missingContainers, start, end)
+	if err != nil {
+		log.Errorf("Error fetching historical pod data: %s", err.Error())
+	}
+	return containerNameCost, err
+}
+
+func getKubeCostData(cache clustercache.ClusterCache, clusterID string) ([]*clustercache.Pod, map[string]map[string][]string, map[string]map[string][]string, map[string]map[string]string, map[string]map[string]string, error) {
```

Comment:
This seems unnecessary. Returning 5 maps doesn't feel like it made the functionality more concise or improved clarity. Recommendation here is to move the contents back inline to the original callsite.